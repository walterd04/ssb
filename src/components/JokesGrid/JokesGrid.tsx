import React from "react";
import styled from "styled-components";

import { Joke } from "../../types";

import { JokeCard } from "../JokeCard";

interface JokesGridProps {
  jokes: Joke[];
  updateJokes: React.Dispatch<React.SetStateAction<Joke[]>>;
}

const JokesGridContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-flow: row wrap;
  padding: 20px;
  & > div {
    margin: 20px 20px;
  }
`;

export const JokesGrid: React.FC<JokesGridProps> = ({
  jokes,
  updateJokes,
}: JokesGridProps) => {
  const updateJoke = (joke: Joke) => {
    const updatedJokes = jokes.map((existingJoke: Joke) => {
      if (joke.id === existingJoke.id) {
        return joke;
      }

      return existingJoke;
    });

    updateJokes(updatedJokes);
  };

  return (
    <JokesGridContainer>
      {jokes.map((joke: Joke) => (
        <JokeCard key={joke.id} joke={joke} updateJoke={updateJoke} />
      ))}
    </JokesGridContainer>
  );
};
