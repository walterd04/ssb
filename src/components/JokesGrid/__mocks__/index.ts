import { Joke } from "../../../types";

export const jokes: Joke[] = [
  {
    category: "Misc",
    type: "twopart",
    setup:
      "Arguing with a woman is like reading a software's license agreement.",
    delivery: 'In the end you ignore everything and click "I agree".',
    flags: {
      nsfw: false,
      religious: false,
      political: false,
    },
    id: 71,
    safe: false,
    lang: "en",
  },
  {
    category: "Pun",
    type: "twopart",
    setup: "What do you call a pile of kittens?",
    delivery: "A meowntain.",
    flags: {
      nsfw: false,
      religious: false,
      political: false,
    },
    id: 130,
    safe: true,
    lang: "en",
  },
];
