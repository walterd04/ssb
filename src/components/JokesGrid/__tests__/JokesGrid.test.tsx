import React from "react";
import { JokesGrid } from "../JokesGrid";
import { render } from "@testing-library/react";
// eslint-disable-next-line jest/no-mocks-import
import { jokes } from "../__mocks__";

describe("UI Tests", () => {
  it("Should render without crashing, matches snapshot", () => {
    const { baseElement } = render(
      <JokesGrid jokes={jokes} updateJokes={jest.fn()} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("Should render the correct joke category", () => {
    const { getByText } = render(
      <JokesGrid jokes={jokes} updateJokes={jest.fn()} />
    );

    expect(getByText("Misc")).toBeTruthy();
    expect(getByText("Pun")).toBeTruthy();
  });
});
