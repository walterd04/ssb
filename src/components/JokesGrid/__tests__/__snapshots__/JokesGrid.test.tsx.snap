// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Tests Should render without crashing, matches snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-jrsJWt bXuSqb"
    >
      <div
        class="sc-bdnxRM kUkMNW"
      >
        <div
          class="sc-gtsrHT KZcwU"
        >
          <h2
            class="sc-dlnjwi dknzix"
          >
            Misc
          </h2>
        </div>
        <div
          class="sc-gtsrHT KZcwU"
        >
          <img
            alt="red x mark"
            class="sc-eCApnc hKxami"
            src="unsafe.svg"
            title="not safe"
          />
          <div
            class="sc-gKAaRy fJEFyC"
          >
            <p
              class="sc-jSFjdj bEexGY"
            >
              Arguing with a woman is like reading a software's license agreement.
            </p>
            <p
              class="sc-jSFjdj bEexGY"
            >
              In the end you ignore everything and click "I agree".
            </p>
          </div>
        </div>
        <button
          class="sc-fujyAs fnDpAK"
          type="button"
        >
          Edit Joke
        </button>
      </div>
      <div
        class="sc-bdnxRM kUkMNW"
      >
        <div
          class="sc-gtsrHT KZcwU"
        >
          <h2
            class="sc-dlnjwi dknzix"
          >
            Pun
          </h2>
        </div>
        <div
          class="sc-gtsrHT KZcwU"
        >
          <img
            alt="green checkmark"
            class="sc-eCApnc hKxami"
            src="safe.svg"
            title="safe"
          />
          <div
            class="sc-gKAaRy fJEFyC"
          >
            <p
              class="sc-jSFjdj bEexGY"
            >
              What do you call a pile of kittens?
            </p>
            <p
              class="sc-jSFjdj bEexGY"
            >
              A meowntain.
            </p>
          </div>
        </div>
        <button
          class="sc-fujyAs fnDpAK"
          type="button"
        >
          Edit Joke
        </button>
      </div>
    </div>
  </div>
</body>
`;
